src/Plugins/ComplexCore/src/ComplexCore/Filters/RotateSampleRefFrame.cpp

```diff
@@ -0,0 +1,540 @@
+  }
+  }
+}
+
+Result<Matrix3fR> ConvertAxisAngleToRotationMatrix(const std::vector<float32>& rotationAxisVec, float32 rotationAngle)
+{
+  const Eigen::Vector3f rotationAxis(rotationAxisVec.data());
+  float32 norm = rotationAxis.norm();
+  std::vector<Warning> warnings;
+  if(!closeEnough(norm, 1.0f, k_Threshold))
+  {
+    warnings.push_back(Warning{-45003, fmt::format("Axis angle is not normalized (norm is {}). Filter will automatically normalize the value.", norm)});
+  }
+
+  float32 rotationAngleRadians = rotationAngle * (numbers::pi / 180.0);
+
+  Eigen::AngleAxisf axisAngle(rotationAngleRadians, rotationAxis.normalized());
+
+  Matrix3fR rotationMatrix = axisAngle.toRotationMatrix();
+
+  return {rotationMatrix, std::move(warnings)};
+}
+
+Result<Matrix3fR> ConvertRotationTableToRotationMatrix(const std::vector<std::vector<float64>>& rotationMatrixTable)
+{
+  if(rotationMatrixTable.size() != 3)
+  {
+    return MakeErrorResult<Matrix3fR>(-45004, "Rotation Matrix must be 3 x 3");
+  }
+
+  for(const auto& row : rotationMatrixTable)
+  {
+    if(row.size() != 3)
+    {
+      return MakeErrorResult<Matrix3fR>(-45005, "Rotation Matrix must be 3 x 3");
+    }
+  }
+
+  Matrix3fR rotationMatrix = ConvertTableToMatrix(rotationMatrixTable);
+
+  float32 determinant = rotationMatrix.determinant();
+
+  if(!closeEnough(determinant, 1.0f, k_Threshold))
+  {
+    return MakeErrorResult<Matrix3fR>(-45006, fmt::format("Rotation Matrix must have a determinant of 1 (is {})", determinant));
+  }
+
+  Matrix3fR transpose = rotationMatrix.transpose();
+  Matrix3fR inverse = rotationMatrix.inverse();
+
+  if(!transpose.isApprox(inverse, k_Threshold))
+  {
+    return MakeErrorResult<Matrix3fR>(-45007, "Rotation Matrix's inverse and transpose must be equal");
+  }
+
+  return {rotationMatrix};
+}
+
+Result<Matrix3fR> ComputeRotationMatrix(const Arguments& args)
+{
+  auto rotationRepresentationIndex = args.value<uint64>(RotateSampleRefFrame::k_RotationRepresentation_Key);
+
+  RotationRepresentation rotationRepresentation = CastIndexToRotationRepresentation(rotationRepresentationIndex);
+
+  switch(rotationRepresentation)
+  {
+  case RotationRepresentation::AxisAngle: {
+    auto rotationAxisVec = args.value<std::vector<float32>>(RotateSampleRefFrame::k_RotationAxis_Key);
+    auto rotationAngle = args.value<float32>(RotateSampleRefFrame::k_RotationAngle_Key);
+
+    return ConvertAxisAngleToRotationMatrix(rotationAxisVec, rotationAngle);
+  }
+  case RotationRepresentation::RotationMatrix: {
+    auto rotationMatrixTableData = args.value<DynamicTableData>(RotateSampleRefFrame::k_RotationMatrix_Key);
+    auto rotationMatrixTable = rotationMatrixTableData.getTableData();
+
+    return ConvertRotationTableToRotationMatrix(rotationMatrixTable);
+  }
+  default: {
+    throw std::runtime_error("RotateSampleRefFrame: Unsupported RotationRepresentation");
+  }
+  }
+}
+
+struct CopyDataFunctor
+{
+  template <class T>
```

Comment:
This can/should be parallelized? Seems like we are copying from an old array into a brand new array so that parallelization should be straight forward.